tests/test_level/test_level_12.py

```diff
@@ -282,6 +282,160 @@ def test_print_calc(self):
 
         self.multi_level_tester(code=code, expected=expected, max_level=17)
 
+    #
+    # forward tests
+    #
+    def test_forward_with_integer_variable(self):
+        code = textwrap.dedent("""\
+            a is 50
+            forward a""")
+        expected = HedyTester.dedent(
+            "a = 50",
+            HedyTester.forward_transpiled('a'))
+
+        self.multi_level_tester(
+            code=code,
+            expected=expected,
+            extra_check_function=self.is_turtle(),
+            max_level=18,
+        )
+
+    def test_forward_with_string_variable_gives_type_error(self):
+        code = textwrap.dedent("""\
+            a is "ten"
+            forward a""")
+
+        self.multi_level_tester(
+            code=code,
+            exception=hedy.exceptions.InvalidArgumentTypeException,
+            max_level=18,
+        )
+
+    def test_forward_with_list_access_random(self):
+        code = textwrap.dedent("""\
+        directions is 10, 100, 360
+        forward directions at random""")
+
+        expected = HedyTester.dedent("""\
+        directions = [10, 100, 360]""",
+        HedyTester.forward_transpiled('random.choice(directions)'))
+
+        self.multi_level_tester(
+            max_level=15,
```

Comment:
If max_level is less than 18 (HEDY_MAX_LEVEL) we should add tests for the higher levels too. So in this case I would expect a similar test at level 16.

Reply:
I've added some tests regarding lists in level 16. Lists are changed there, so new tests are required.